### DISCUSSION TITLE
Recognition Confidence metric

### DISCUSSION THREAD

**Discussion by daviembrito** [PaddlePaddle/PaddleOCR#11352] [Q&A]
Hi!

Does someone know what metric and formula is used to calcule recognition confidence for character recognition? Didn't find anything around here. Thanks!

**Comment by SimWalther** [comment]
Hi, I'm interested in knowing too. I searched in the documentation and I couldn't find anything.
I looked at the PP-OCRv3 ([PP-OCRv3: More Attempts for the Improvement of Ultra Lightweight OCR System](https://arxiv.org/pdf/2206.03001.pdf)) paper and I couldn't find mention of confidence but I found out that the text recognition model  (SVTR-LCNet) make use of the [CTC](https://www.cs.toronto.edu/~graves/icml_2006.pdf) loss but guided by an attention mechanism ([GTC](https://arxiv.org/pdf/2002.01276.pdf)).

The CTC paper talks about the total probability of the label sequences:
"A CTC network has a softmax output layer (Bridle, 1990) with one more unit than there are labels in L. The activations of the first |L| units are interpreted as the probabilities of observing the corresponding labels at particular times. The activation of the extra unit is the probability of observing a ‘blank’, or no label. Together, these outputs define the probabilities of all possible ways of aligning all possible label sequences with the input sequence. **The total probability of any one label sequence can then be found by summing the probabilities of its different alignments.**" (Connectionist Temporal Classification: Labelling Unsegmented Sequence Data with Recurrent Neural Networks)

I also found an example but with RCNN where the CTC output probability is used as the confidence ([Dive into OCR Text Recognition Practice](https://aistudio.baidu.com/aistudio/projectdetail/3449403?channelType=0&channel=0)):

"
When the network is randomly initialized, the output results are unordered. Through SoftMax, the prediction results with the highest probability at each time step can be obtained, where: pred_id represents the predicted tag ID, and pre_scores represents the confidence of the predicted result:

```
ctc_head = CTCHead(in_channels=96, out_channels=37)
predict = ctc_head(sequence)
print("predict shape:", predict.shape)
result = F.softmax(predict, axis=2)
pred_id = paddle.argmax(result, axis=2)
pred_socres = paddle.max(result, axis=2)
print("pred_id:", pred_id)
print("pred_scores:", pred_socres)
```
"


So I think that the confidence is the output label sequences softmax probabilities but I may be wrong.
